llm_gpt4all.py

```diff
@@ -108,7 +112,13 @@ def execute(self, prompt, stream, response, conversation):
             if system:
                 text_prompt = f"{system}\n{text_prompt}"
             response.response_json = {"full_prompt": text_prompt}
-            gpt_model = GPT4All(self.filename())
+
+            # We assume file existing is enough to enable download, does not check if file is complete
```

Comment:
We could take the check even further by computing the md5 checksum, and seeing if it matches before resuming.

https://raw.githubusercontent.com/nomic-ai/gpt4all/main/gpt4all-chat/metadata/models2.json